packages/contracts-bedrock/test/L1/SuperchainConfig.t.sol

```diff
@@ -105,3 +133,186 @@ contract SuperchainConfig_Unpause_Test is CommonTest {
+            bytes32(uint256(uint160(_portal)))
+        );
+        vm.expectCall(address(systemConfig), abi.encodeWithSelector(ISystemConfigInterop.optimismPortal.selector));
+
+        // Mock and expect the call to authorize the portal on the SharedLockbox with the `_portal` address
+        _mockAndExpect(
+            address(sharedLockbox), abi.encodeWithSelector(ISharedLockbox.authorizePortal.selector, _portal), ""
+        );
+
+        // Expect the ChainAdded event to be emitted
+        vm.expectEmit(address(superchainConfigForTest));
+        emit ChainAdded(_chainId, address(systemConfig), _portal);
+
+        // Add the new chain to the dependency set
+        vm.prank(superchainConfigForTest.guardian());
+        superchainConfigForTest.addChain(_chainId, address(systemConfig));
+
+        // Check that the new chain is in the dependency set
+        assertTrue(superchainConfigForTest.isInDependencySet(_chainId));
+    }
+}
+
+contract SuperchainConfig_IsInDependencySet_Test is CommonTest {
+    /// @dev Tests that `isInDependencySet` returns false when the chain is not in the dependency set. Checking if empty
+    ///      to ensure that should be true.
+    function test_isInDependencySet_false_succeeds(uint256 _chainId) external view {
+        assert(superchainConfig.dependencySet().length == 0);
+        assertFalse(superchainConfig.isInDependencySet(_chainId));
+    }
+
+    /// @dev Tests that `isInDependencySet` returns true when the chain is in the dependency set.
+    function test_isInDependencySet_true_succeeds(uint256 _chainId) external {
+        SuperchainConfigForTest superchainConfig = new SuperchainConfigForTest(address(sharedLockbox));
+        superchainConfig.forTest_addChainOnDependencySet(_chainId);
+
+        assertTrue(superchainConfig.isInDependencySet(_chainId));
+    }
+}
+
+contract SuperchainConfig_DependencySet_Test is CommonTest {
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    EnumerableSet.UintSet internal chainIds;
+
+    /// @notice Tests that the dependency set returns properly the dependencies added.
+    function test_dependencySet_succeeds(uint256[] calldata _chainIdsArray) public {
+        SuperchainConfigForTest superchainConfigForTest = new SuperchainConfigForTest(address(sharedLockbox));
+
+        // Ensure there are no repeated values on the input array
+        for (uint256 i; i < _chainIdsArray.length; i++) {
+            chainIds.add(_chainIdsArray[i]);
+        }
+
+        // Add the dependencies to the dependency set
+        for (uint256 i; i < chainIds.values().length; i++) {
+            superchainConfigForTest.forTest_addChainOnDependencySet(chainIds.at(i));
+        }
+
+        // Check that the dependency set has the same length as the dependencies
+        uint256[] memory dependencySet = superchainConfigForTest.dependencySet();
+        assertEq(dependencySet.length, chainIds.values().length);
```

Comment:
```suggestion
        assertEq(dependencySet.length, chainIds.length());
```

Reply:
Not possible since is an enumerable set (struct under the hood) and not an array. You need to query the array first to get the length

Reply:
then how is it possible to use it like that on `SuperchainConfig.sol` [here](https://github.com/defi-wonderland/optimism/pull/138/files#diff-016d692e745a10a098159957d937a3a02c5317f6bf727d24c584f8448e81cfceR136)

Reply:
my bad, pushed here https://github.com/defi-wonderland/optimism/pull/138/commits/5e7e2cd409e0d45cf7c2719b2ef10fb33ef7dd12